### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Anton-Zhytonbaiev**
[DEMO LINK](https://Anton-Zhytonbaiev.github.io/react_keyboard/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    message: 'Nothing was pressed yet',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ message: `The last pressed key is ${event.key}` });
```
Comment (GarryRocksman) [inline src/App.tsx:14]: think you will need to get a key value for another function to operate, from this state it won't be easy to manage where key value

**Review by GarryRocksman — CHANGES_REQUESTED** [review]
GJ 🔥 but needs some fixes

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.keyupHandler);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.keyupHandler);
+  }
+
+  keyupHandler = (event: KeyboardEvent) => {
```
Comment (ArturErmolenko) [inline src/App.tsx:20]: checklist point 1

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,39 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  message: string | null;
```
Comment (ArturErmolenko) [inline src/App.tsx:4]: it's not clear to me what message , I am sure that this variable should be named differently
```suggestion
  pressedKey: string;
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,39 @@
...
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    message: null,
```
Comment (ArturErmolenko) [inline src/App.tsx:9]:
```suggestion
    message: '',
```

**Review by ArturErmolenko — CHANGES_REQUESTED** [review]
GJ 👍 
Left comments

**Review by ArturErmolenko — APPROVED** [review]
Well done 👍

**Pull request closed without merging**